### PR TITLE
added forward test

### PR DESCRIPTION
**Pull request by aanurraj**
## Description
added forward test.

## Affected Dependencies
none

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by gmuraru** [comment]
Shouldn't the codecov be published on the PR?

File: tests/sympc/tensor/grad/grad_functions_test.py [hunk 1]
```diff
@@ -561,3 +562,44 @@ def test_grad_matmul_backward(get_clients) -> None:
...
+    out_torch = model(x_secret)
+    out_mpc = mpc_model(x_mpc)
+
+    s_torch = torch.sum(out_torch)
+    s_mpc = out_mpc.sum()
+
+    s_torch.backward()
+    s_mpc.backward()
+
+    assert np.allclose(x_mpc.grad.get(), x_secret.grad, rtol=1e-3)
```
Comment (Syzygianinfern0) [inline tests/sympc/tensor/grad/grad_functions_test.py:605]: Could u locally put this in a for loop for 30 times and check if it passes every time? I have a feeling this test is going to be a flaky one 😂

**Review by Syzygianinfern0 — APPROVED** [review]
😍

**Comment by aanurraj** [comment]
> Shouldn't the codecov be published on the PR?

yes ideally it should, I am not sure why it comments on some PRs only.

File: tests/sympc/tensor/grad/grad_functions_test.py (hunk 1 above)
Comment (aanurraj) [reply tests/sympc/tensor/grad/grad_functions_test.py:605]: hahah, will try :P

Comment (Syzygianinfern0) [reply tests/sympc/tensor/grad/grad_functions_test.py:605]: 😆 Cool, let us know what happens 👉👈

Comment (gmuraru) [reply tests/sympc/tensor/grad/grad_functions_test.py:605]: Looks good @aanurraj . After you run it multiple times and it is not flaky we can merge this, I think

**Review by gmuraru — APPROVED** [review]
